sqs_workers/memory_sqs.py

```diff
@@ -158,19 +160,40 @@ def delete_messages(self, Entries):
         message_ids = {entry["Id"] for entry in Entries}
 
         successfully_deleted = set()
-        push_back_messages = []
 
-        for message in self.messages:
+        for i, message in enumerate(self.in_flight):
             if message.message_id in message_ids:
                 successfully_deleted.add(message.message_id)
-            else:
-                push_back_messages.append(message)
-        self.messages[:] = push_back_messages
+                del self.in_flight[i]
 
-        didnt_deleted = message_ids.difference(successfully_deleted)
         return {
             "Successful": [{"Id": _id} for _id in successfully_deleted],
-            "Failed": [{"Id": _id} for _id in didnt_deleted],
+        }
+
+    def change_message_visibility_batch(self, Entries):
+        """
+        Changes message visibility by looking at in-flight messages, setting
+        a new execute_at, and returning it to the pool of processable messages
+        """
```

Comment:
Technically, this doesn't change a message's visibility, right? Instead, it takes in-flight messages that match entries passed-in as an argument and sets their execution time to happen in `VisibilityTimeout` seconds, correct? Perhaps the function name and documentation should reflect that? Something like `schedule_messages` or even `schedule_messages_after_visibility_timeout`?

Reply:
Yes you're spot on! I think `schedule_messages` would definitely be a better name here. But, it's a bit complicated for two reasons:

* It only seems here like we're "scheduling messages for the future" here because it's how the toy impl decided to mock SQS's behavior -- this isn't how SQS really works, it's just close enough, right. For real SQS, it isn't saying the message can't be processed until X date, just that it won't be visible (i.e. turn up in responses to `receive` calls from other workers) until X date, which is a bit different
* It actually MUST be named `change_message_visibility_batch` anyway! Since the purpose of `MemoryQueue` is to act like a real SQS API in tests/etc, and that's what the real API (and thus boto) call the function 😛 

It is a little unsatisfying lol! But does it sound reasonable to you?

Reply:
Oh yeah, that makes a lot of sense. It HAS to be named this way, so no discussion at all. 😄